### PR TITLE
README.rst: tidy up redundant instructions

### PR DESCRIPTION
**Pull request by zadacka**
Remove the README section which detailed how to force `psycopg2` 2.7 dependency installation from source rather than binary.

`psycopg2` 2.7 would automatically favour installation from binary. A bug affecting some users made it desirable to have the option to install from source instead. The README detailed how to do this using pip's `--no-binary` option (nice!).

`psycopg2` 2.8 has now split the binary and source dependencies into different packages with different names (source: [psycopg2 docs](http://initd.org/psycopg/docs/install.html#change-in-binary-packages-between-psycopg-2-7-and-2-8)) which makes the `--no-binary` instructions redundant. To specify source/binary you now need to explicitly specify the desired package.

As of #1060, installing `pgcli` from pip will result in the installation of `psycopg2` 2.8 from source.

## Checklist

- [x] I've added this contribution to the `changelog.rst`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

- [ ] I installed pre-commit hooks (`pip install pre-commit && pre-commit install`), and ran `black` on my code.

### CODE REVIEW

**Comment by zadacka** [comment]
@j-bennet - thanks! I've added some extra description to the README for psycopg2 2.7.x users, and updated changelog and AUTHORS ... and changelog again to actually link my name to a  URL.

However, my experience is that pgcli won't work if we attempt to meet the psycopg2 dependency by installing from a binary.  For example, given the setup below:
```
➜  pip freeze | grep psycopg2
psycopg2-binary==2.8.2
```

Even though python can see it:
```
➜  python
>>> import psycopg2
>>>
```

Attempting to run pgcli still errors:
```
➜  pgcli
Traceback (most recent call last):
  File "/opt/bats/bin/pgcli", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/opt/bats/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2991, in <module>
    @_call_aside
  File "/opt/bats/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2977, in _call_aside
    f(*args, **kwargs)
  File "/opt/bats/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3004, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/opt/bats/lib/python2.7/site-packages/pkg_resources/__init__.py", line 662, in _build_master
    ws.require(__requires__)
  File "/opt/bats/lib/python2.7/site-packages/pkg_resources/__init__.py", line 970, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/opt/bats/lib/python2.7/site-packages/pkg_resources/__init__.py", line 856, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'psycopg2>=2.7.4' distribution was not found and is required by pgcli
```

Which makes sense to me, given that the source and binary distributions have distinct names. 
It's always possible that I've got something strange with my configuration. Does your experience differ here?

**Comment by j-bennet** [comment]
Very interesting. It looks like this part is platform-dependent. I get the same `DistributionError` as you on a Mac, but not on my Pixelbook (Debian Linux). On Linux, I had to jump through a hoop to make things work, but they do:

```
(pgcli) --- src/pgcli ‹master* ?› » pip freeze | grep psycopg2 
psycopg2-binary==2.8.2
(pgcli) --- src/pgcli ‹master* ?› » 
(pgcli) --- src/pgcli ‹master* ?› » pgcli postgres postgres                                                        
Server: PostgreSQL 9.6.13
Version: 2.1.0
Chat: https://gitter.im/dbcli/pgcli
Mail: https://groups.google.com/forum/#!forum/pgcli
Home: http://pgcli.com
postgres>                                                                                                                     
Goodbye!
```

What I had to do was:

1. `pip uninstall psycopg2`
2. `rm -rf ~/.pyenv/versions/3.6.5/envs//pgcli/lib/python3.6/site-packages/psycopg2`
3. `pip install psycopg2-binary`

The "hoop" is step 2. Without this step, I had an error starting pgcli:

```
(pgcli) --- src/pgcli ‹master* ?› » pgcli postgres postgres
Traceback (most recent call last):
  File "/home/ichernyavska/.pyenv/versions/pgcli/bin/pgcli", line 6, in <module>
    from pgcli.main import cli
  File "/home/ichernyavska/src/pgcli/pgcli/main.py", line 55, in <module>
    from .pgexecute import PGExecute
  File "/home/ichernyavska/src/pgcli/pgcli/pgexecute.py", line 4, in <module>
    import psycopg2.extras
ModuleNotFoundError: No module named 'psycopg2.extras'
```

Bottom line, it looks like installing from binary will indeed not work, at least on some platforms. So definitely better leave that out of README. I am happy with this documentation update.

**Comment by j-bennet** [comment]
Merged. Thank you for the PR! 🍒 Sorry about the confusion with https://github.com/dbcli/pgcli/pull/1059.

**Comment by zadacka** [comment]
My pleasure! Kudos on running such a good project! I had been using `psql` daily, and `pgcli` is *so* much more friendly.
